src/app/component/Draggable/Draggable.component.js

```diff
@@ -36,6 +38,30 @@ class Draggable extends Component {
         this.handleTouchEnd = this.handleTouchEnd.bind(this);
     }
 
+    static getDerivedStateFromProps(props, state) {
+        if (props.draggableRef.current !== null) {
+            const { activeSlide, isVertical } = props;
+            const { prevActiveSlider } = state;
+            const slideSize = isVertical
+                ? props.draggableRef.current.clientHeight
+                : props.draggableRef.current.clientWidth;
+
+            if (prevActiveSlider !== activeSlide) {
+                if (isVertical) {
```

Comment:
Dragable should not know anything about `activeSlide` nor `isVertical` it should just update values of translate. Single responsibility - remember!